java/src/com/ibm/streamsx/topology/generator/spl/OperatorGenerator.java

```diff
@@ -211,37 +211,58 @@ private void viewAnnotation(JsonObject op, StringBuilder sb) {
 
     private void parallelAnnotation(JsonObject op, StringBuilder sb) {
         boolean parallel = jboolean(op, "parallelOperator");
-
+        
         if (parallel) {
+            boolean partitioned = jboolean(op, "partitioned");
+            JsonObject parallelInfo = op.get("parallelInfo").getAsJsonObject();
+            
             sb.append("@parallel(width=");
-            JsonElement width = op.get("width");
+            JsonElement width = parallelInfo.get(OpProperties.WIDTH);
             if (width.isJsonPrimitive()) {
                 sb.append(width.getAsString());
             } else {
                 splValueSupportingSubmission(width.getAsJsonObject(), sb);
             }
-            String parallelInputPortName = jstring(op, "parallelInputPortName");
-            boolean partitioned = jboolean(op, "partitioned");
+      
             if (partitioned) {
+                sb.append(", partitionBy=[");
+                JsonArray partitionedPorts = array(parallelInfo, "partitionedPorts");
+                for(int i = 0; i < partitionedPorts.size(); i++){
+                    JsonObject partitionedPort = partitionedPorts.get(i).getAsJsonObject();
+                    
+                    if(i>0)
+                        sb.append(", ");
+                    
+                    sb.append("{port=");
+                    sb.append(getSPLCompatibleName(GsonUtilities.jstring(partitionedPort, "name")));
+                    sb.append(", attributes=[");
+                    JsonArray partitionKeys = partitionedPort.get("partitionedKeys").getAsJsonArray();
+                    for (int j = 0; j < partitionKeys.size(); j++) {
+                        if (j != 0)
+                            sb.append(", ");
+                        sb.append(partitionKeys.get(j).getAsString());
+                    }
+                    sb.append("]}");
+                }
                 
-                JsonArray partitionKeys = op.get("partitionedKeys").getAsJsonArray();
-
-                parallelInputPortName = getSPLCompatibleName(parallelInputPortName);
-                sb.append(", partitionBy=[{port=");
-                sb.append(parallelInputPortName);
-                sb.append(", attributes=[");
-                for (int i = 0; i < partitionKeys.size(); i++) {
+                sb.append("]"); 
+            }
+            
+            JsonArray broadcastPorts = parallelInfo.get("broadcastPorts").getAsJsonArray();
+            if(broadcastPorts.size() > 0){
```

Comment:
Minor nit - use of `!broadcastPorts.isEmpty()` method is clearer.

Reply:
JsonArray doesn't have an `isEmpty` method.